### PR TITLE
Forms with client-side errors

### PR DESCRIPTION
**Pull request by cirdes**

### CODE REVIEW

File: app/views/components/rbui/button/button.rb [hunk 1]
```diff
@@ -88,7 +88,6 @@ def default_classes
 
     def default_attrs
       {
-        type: "button",
```
Comment (pierry01) [inline app/views/components/rbui/button/button.rb:91]: oops! This is so important! Please do not remove it

File: app/views/components/rbui/form/form.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    end
+
+    private
+
+    def default_attrs
+      {
+        novalidate: true,
+        data: {
+          controller: "rbui--form",
+          action: "rbui--form#onSubmit",
```
Comment (pierry01) [inline app/views/components/rbui/form/form.rb:16]: it should be `submit->rbui--form#onSubmit`?

File: app/views/components/rbui/form/form_field.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      div(**attrs, &)
+    end
+
+    private
+
+    def default_attrs
+      {
+        data: {
+          controller: "rbui--form-field",
+          rbui__form_field_dirty_value: "false"
```
Comment (pierry01) [inline app/views/components/rbui/form/form_field.rb:15]: I think you can pass the boolean value here instead of a string

**Review by pierry01 — APPROVED** [review]

File: app/views/components/rbui/form/form_field.rb (hunk 3 above)
Comment (cirdes) [reply app/views/components/rbui/form/form_field.rb:15]: @pierry01 , I like to keep everything on the right side as a string.

Stimulus encode this as String: https://stimulus.hotwired.dev/reference/values#types

**Pull request merged**